pluralsight/design-system
# actionmenu: focus and link color fixes

### What You're Solving

- Don't shows first item as focused if shouldFocusOnMount is false
- Prevent active item from stealing focus when menu is open and user clicks
on another focusable element
- Prevent two items from looking like they are both focused (one has actual
focus, the other is just styled). Happens when activeIndex is not kept in sync
with the actually focused item.
- Prevent global anchor tag styles from changing the color of menu items

##### How to Verify

- Add a global style `a:hover { color: red; }` to the page and verify that hovered items with an href prop do not change to red.
- Verify that a menu with `shouldFocusOnMount={false}` does not add a gray background to the first element.
- Navigate through the menu with tabs and verify you can tab out from the top and bottom of the list
- Add an actionMenu and an input to a page and verify that you can focus the input (actionMenu was always stealing focus on `componentDidUpdate` before)
